Real-Dev-Squad/feature-flag-backend
# Added local DynamoDB support

Closes #60 

Steps to test:
- Download and setup DynamoDB locally following the steps mentioned in [Download DynamoDB local v2.x](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBLocal.DownloadingAndRunning.html)
- Run command 
```
make dev_db_setup DYNAMODB_LOCAL_ROOT_FOLDER_PATH={Pass the root folder path of DynamoDBLocal.jar file downloaded in above step with a trailing path separator}
```
(For windows use `//` as path separator in above make command)
 - Add the replace directive above require lines in `go.mod` file of the lambda function you want to test
 ```go
 replace "github.com/Real-Dev-Squad/feature-flag-backend" => ../
```
- Run this command 
`rm -rf .aws-sam/ && sam build && sam local start-api`
- Hit the endpoint you want to test
Eg. http://localhost:3000/health-check
